### PR TITLE
Check for nan in OFTTime in OGRFeature::GetFieldAsString

### PR DESCRIPTION
**Pull request by schwehr**
Found with autofuzz on the CSV driver

### CODE REVIEW

**Comment by schwehr** [comment]
Example failures... ?
```
  TEST: netcdf_45 ... WKT,int32,int32_explicit_fillValue,float64,float64_explicit_fillValue,string1char,string3chars,twodimstringchar,date,datetime_explicit_fillValue,datetime,int64var,int64var_explicit_fillValue,boolean,boolean_explicit_fillValue,float32,float32_explicit_fillValue,int16,int16_explicit_fillValue,x,byte_field
"POINT Z (1 2 3)","1","1",1.23456789012,1.23456789012,x,STR,STR,1970/01/02,2016/02/06 12:34:56.789,2016/02/06 12:34:56.789,"1234567890123","1234567890123","1","1",1.2,1.2,"123","12",5,"-125"
"POINT (1 2)",,,,,,,,,,,,,,,,,,,,
,,,,,,,,,,,,,,,,,,,,

Script: ogr/ogr_ods.py
  TEST: ogr_ods_6 ... fail
    line 380: did not get expected result: [u'Field1,Field2,Field3,Field4,Field5,Field6,Field7,Field8,Field9,Field10,Field11,Field12,Field13,Field14,Field15,Field16,Field17,Field18,Field19,Field20,Field21,Field22,Field23,Field24,Field25,Field26,Field27,Field28,Field29,Field30,Field31,Field32', u'of:=[.B1],of:=[.C1],of:=[.A1],,,,,,,,,,,,,,,,,,,,,,,,,,,,,', u'"1","1","1",,,,,,,,,,,,,,,,,,,,,,,,,,,,,', u'ab,ab,ab,,,,,,,,,,,,,,,,,,,,,,,,,,,,,', u'"1",a,,"3.5",MIN,"1",MIN,"3.5",SUM,"4.5",AVERAGE,"2.25",COUNT,"2",COUNTA,"3",,,,,,,,,,,,,,,,', u'abcdef,"6",,a,abcdef,,f,abcdef,"of:=MID([.A5];0;1)",,a,abcdef,,a,ef,ef,,,,,,,,,,,,,,,,', u'"1",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,', u'AB,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,', u'"2","2","0","3","1","0","0","1","1","1","0","0","0","1","1","0",,,,,,,,,,,,,,,,', u'"1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1"', u'"0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0"']

```

File: gdal/ogr/ogrfeature.cpp [hunk 1]
```diff
@@ -2443,7 +2443,7 @@ const char *OGRFeature::GetFieldAsString( int iField ) const
     else if( eType == OFTTime )
     {
         const int ms = OGR_GET_MS(pauFields[iField].Date.Second);
-        if( ms != 0 )
+        if( ms != 0 || CPLIsNan(pauFields[iField].Date.Second) )
```
Comment (rouault) [inline gdal/ogr/ogrfeature.cpp]: couldn't you just swith the then and else clause and test ms == 0 ? This would avoid the annoying NaN check

**Comment by rouault** [comment]
The test failures are unrelated and due to one of my today's change. Should now be fixed

File: gdal/ogr/ogrfeature.cpp (hunk 1 above)
Comment (rouault) [reply gdal/ogr/ogrfeature.cpp]: Ah that wouldn't solve the issue, as OGR_GET_MS() returns 0 for NaN. Ok merging thins then

Comment (schwehr) [reply gdal/ogr/ogrfeature.cpp]: Thanks for thinking it through.  These trip me up so often.  This was my 3rd try at a patch for this issue.  My other tries were really ugly.

**Pull request merged**